backpack/core/derivatives/rnn.py

```diff
@@ -264,15 +282,33 @@ def _weight_hh_l0_jac_t_mat_prod(
             product
         """
         self._check_parameters(module)
-        N_axis = get_batch_axis(module, "input0")
+        N_axis, T_axis = self.get_batch_and_time_axes(module)
         N: int = mat.shape[N_axis + 1]
         H: int = mat.shape[3]
         output = subsample(module.output, dim=N_axis, subsampling=subsampling)
         return einsum(
-            "vtnh,tnk->" + ("vhk" if sum_batch else "vnhk"),
-            self._a_jac_t_mat_prod(output, module.weight_hh_l0, mat),
+            f"vtnh,{'nt' if module.batch_first else 'tn'}k->"
+            + ("vhk" if sum_batch else "vnhk"),
+            self._a_jac_t_mat_prod(module, module.weight_hh_l0, mat, subsampling),
             cat(
-                [zeros(1, N, H, device=mat.device, dtype=mat.dtype), output[0:-1]],
-                dim=0,
+                [
+                    zeros(N, 1, H, device=mat.device, dtype=mat.dtype)
+                    if module.batch_first
+                    else zeros(1, N, H, device=mat.device, dtype=mat.dtype),
```

Comment:
`zeros(*shape)` and extract `shape = (...) if batch_first else (...)`

Reply:
I extracted the whole `output_shifted`